fe/fe-core/src/main/java/com/starrocks/external/HiveMetaStoreTableUtils.java

```diff
@@ -84,6 +94,82 @@ public static long getPartitionStatsRowCount(HiveMetaStoreTableInfo hmsTable,
+        String typeUpperCase = Utils.getTypeKeyword(hiveType).toUpperCase();
+        switch (typeUpperCase) {
+            case "TINYINT":
+                return PrimitiveType.TINYINT;
+            case "SMALLINT":
+                return PrimitiveType.SMALLINT;
+            case "INT":
+            case "INTEGER":
+                return PrimitiveType.INT;
+            case "BIGINT":
+                return PrimitiveType.BIGINT;
+            case "FLOAT":
+                return PrimitiveType.FLOAT;
+            case "DOUBLE":
+            case "DOUBLE PRECISION":
+                return PrimitiveType.DOUBLE;
+            case "DECIMAL":
+            case "NUMERIC":
+                return PrimitiveType.DECIMAL32;
+            case "TIMESTAMP":
+                return PrimitiveType.DATETIME;
+            case "DATE":
+                return PrimitiveType.DATE;
+            case "STRING":
+            case "VARCHAR":
+            case "BINARY":
+                return PrimitiveType.VARCHAR;
+            case "CHAR":
+                return PrimitiveType.CHAR;
+            case "BOOLEAN":
+                return PrimitiveType.BOOLEAN;
+            default:
+                throw new DdlException("hive table column type [" + typeUpperCase + "] transform failed.");
+        }
+    }
+
+    // In the first phase of connector, in order to reduce changes, we use `hive.metastore.uris` as resource name
+    // for table of external catalog. The table of external catalog will not create a real resource.
+    // We will reconstruct this part later. The concept of resource will not be used for external catalog in the future.
+    public static HiveTable convertToSRTable(Table hiveTable, String resoureName) throws DdlException {
+        if (hiveTable.getTableType().equals("VIRTUAL_VIEW")) {
+            throw new DdlException("Hive view table is not supported.");
+        }
+
+        Map<String, FieldSchema> allHiveColumns = getAllHiveColumns(hiveTable);
+        List<Column> fullSchema = Lists.newArrayList();
+        for (Map.Entry<String, FieldSchema> entry : allHiveColumns.entrySet()) {
+            FieldSchema fieldSchema = entry.getValue();
+            PrimitiveType srType = convertColumnType(fieldSchema.getType());
+            Column column = new Column(fieldSchema.getName(), ScalarType.createType(srType), true);
+            fullSchema.add(column);
+        }
+
+        // Adding some necessary properties to adapt initialization of HiveTable.
+        Map<String, String> properties = Maps.newHashMap();
+        properties.put(HiveTable.HIVE_DB, hiveTable.getDbName());
+        properties.put(HiveTable.HIVE_TABLE, hiveTable.getTableName());
+        properties.put(HiveTable.HIVE_METASTORE_URIS, resoureName);
+        properties.put(HiveTable.HIVE_RESOURCE, resoureName);
+
+        return new HiveTable(connectorTableIdIdGenerator.getNextId().asInt(), hiveTable.getTableName(),
```

Comment:
we can't use `GlobalStateMgr.getCurrentState().getNextId()` to get id. beacause it will record edit log. If synchronization is performed on a non-master fe node, it will fail.

Reply:
what's the usage of id? will this id conflict with GlobalStateMgr.getCurrentState().getNextId()

Reply:
I don't think connectorTable's id has any effect, how do you think

Reply:
In fact, it's no sense using an Id on connectorTable, but `Table` need an id.